src/test/starlark/testdata/string_misc.sky

```diff
@@ -41,6 +41,18 @@ assert_eq('b$()n$()n$()'.replace('$()', '$($())'), "b$($())n$($())n$($())")
 assert_eq('b\\n\\n\\'.replace('\\', '$()'), "b$()n$()n$()")
 
 assert_eq('banana'.replace('a', 'e', 2), "benena")
+assert_eq('banana'.replace('a', 'e', 0), "banana")
+assert_eq('banana'.replace('a', 'e', -1), "banana")
```

Comment:
Is this right? I would expect benene.

Reply:
Yes, that's a bug in the current implementation, but as Laurent mentioned above that we'd better not include semantic changes in this PR.

Reply:
Ah, so he did.

In general if you identify a bug but don't intend to fix it in the same PR, it's good practice to at least identify the problem with comments (both at the logic and at the faulty assertions). But let's not hold up this PR any further. LGTM